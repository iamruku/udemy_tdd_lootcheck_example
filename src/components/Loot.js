import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/actions';

export class Loot extends Component {
	componentDidMount() {
		this.props.fetchBitcoin();
	}

	computeBitcoin = () => {
		const { bitcoin } = this.props;

		if (Object.keys(bitcoin).length === 0) return '';

		return (
			this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''), 10)
		);
	};

	render() {
		return (
			<div>
				<h3>Bitcoin Balance: {this.computeBitcoin()}</h3>
			</div>
		);
	}
}

const mapStateToProps = ({ balance, bitcoin }) => ({ balance, bitcoin });

export default connect(
	mapStateToProps,
	{ fetchBitcoin }
)(Loot);
