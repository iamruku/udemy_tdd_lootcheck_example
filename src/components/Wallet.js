import React, { Component } from 'react';
import { connect } from 'react-redux';
import { depositBalance, withdrawBalance } from '../actions/actions';

export class Wallet extends Component {
	state = { inputBalance: 0 };

	deposit = () => {
		this.props.depositBalance(this.state.inputBalance);
	};

	withdraw = () => {
		this.props.withdrawBalance(this.state.inputBalance);
	};

	render() {
		return (
			<div>
				<h2 className="balance">Wallet Balance: {this.props.balance}</h2>
				<br />
				<label>Balance&nbsp;</label>
				<input
					type="text"
					className="inp-balance"
					onChange={e =>
						this.setState({ inputBalance: parseFloat(e.target.value) })
					}></input>
				<button className="btn-deposit" onClick={this.deposit}>
					Deposit
				</button>
				<button className="btn-withdraw" onClick={this.withdraw}>
					Withdraw
				</button>
			</div>
		);
	}
}

const mapStateToProps = ({ balance }) => ({ balance });

export default connect(
	mapStateToProps,
	{ depositBalance, withdrawBalance }
)(Wallet);
