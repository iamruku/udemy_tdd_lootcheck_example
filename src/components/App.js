import React from 'react';
import Wallet from './Wallet';
import Loot from './Loot';

export default () => (
	<div>
		<h2>Loot Check</h2>
		<hr></hr>
		<Wallet></Wallet>
		<Loot></Loot>
	</div>
);
