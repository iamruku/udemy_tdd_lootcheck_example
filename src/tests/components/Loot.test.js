import React from 'react';
import { mount, shallow } from 'enzyme';
import '../setupTests';
import { Loot } from '../../components/Loot';

describe('Loot', () => {
	const _fetchBitcoin = jest.fn();
	const props = { balance: 10, bitcoin: {} };
	let wrapper = shallow(<Loot {...props}></Loot>);

	describe('initial dom content', () => {
		it('renders wrapper', () => {
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('on mount', () => {
		beforeEach(() => {
			props.fetchBitcoin = _fetchBitcoin;
			wrapper = mount(<Loot {...props}></Loot>);
		});

		it('dispatches the `fetchBitcoin()` method via props', () => {
			expect(_fetchBitcoin).toHaveBeenCalled();
		});
	});

	describe('when there is valid bitcoin prop', () => {
		beforeEach(() => {
			const props = {
				balance: 10,
				bitcoin: { bpi: { USD: { rate: '1,000' } } }
			};
			wrapper = shallow(<Loot {...props}></Loot>);
		});

		it('displays the correct bitcoin value', () => {
			expect(wrapper.find('h3').text()).toEqual('Bitcoin Balance: 0.01');
		});
	});
});
