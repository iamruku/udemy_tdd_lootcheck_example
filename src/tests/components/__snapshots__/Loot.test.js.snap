// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot initial dom content renders wrapper 1`] = `ShallowWrapper {}`;
